File: client/src/app/models/order.ts
export interface ShippingAddress {
  fullName: string;
  address1: string;
  address2: string;
  city: string;
  state: string;
  country: string;
  zipCode: string;
}

export interface OrderItem {
  productId: number;
  name: string;
  pictureUrl: string;
  price: number;
  quantity: number;
}

export interface Order {
  id: number;
  buyerId: string;
  shippingAddress: ShippingAddress;
  orderDay: string;
  orderItems: OrderItem[];
  subTotal: number;
  deliveryFee: number;
  orderStatus: string;
  total: number;
}

File: API/Entities/OrderAggregate/Order.cs
namespace Entity.OrderAggregate;

using System.ComponentModel.DataAnnotations;

public class Order {
    public int Id { get; set; }
    public string BuyerId { get; set; }
    [Required]
    public ShippingAddress ShippingAddress { get; set; }
    public DateTime OrderDay { get; set; } = DateTime.Now;
    public List<OrderItem> OrderItems { get; set; }
    public long SubTotal { get; set; }
    public long DeliveryFee { get; set; }
    public OrderStatus OrderStatus { get; set; } = OrderStatus.Pending;
    // for Stripe
    public string PaymentIntentId { get; set; }

    public long GetTotal() {
        return SubTotal + DeliveryFee;
    }
}
